### ISSUE TITLE
TextureProgress node does not display progress when value is set in code (HTML + C# Export only)

### ISSUE THREAD

**Issue by thes3m** [godotengine/godot#47898]
**Godot version:**
3.3 RC8. RC9 x64 Mono (this issue did not occur in previous versions)

** Platform **
Windows (HTML5 Export)

**Issue description:**
TextureProgress node does not display correct progress when progress value is set in code. The progress bar is displayed when game starts, however as soon as `TextureProgress.Value` property is set the progress bar will disapear. This only occurs in HTML5 export (can be runned from editor).

**Steps to reproduce:**
1. Open the provided project
2. Go to Project->Export and add HTML5 export
2. Build and start a desktop project from editor
3. You should see the progress bar filling itself constantly
4. Press HTML5 button in top right corner of editor to run the project in the browser
5. You will see that the progress bar is not displayed on HTML5

** Expected result **
TextureProgress should display progress on HTML5

[GodotProgressBarTest.zip](https://github.com/godotengine/godot/files/6313471/GodotProgressBarTest.zip)

**Comment by Calinou** [comment]
Can you reproduce this in a GDScript project?

**Comment by thes3m** [comment]
@Calinou  No, apparently it only occurs in C#.

**Comment by akien-mga** [comment]
I can confirm the bug with the provided project, and indeed, surprisingly it only happens with C#.

Equivalent code for GDScript:
```gdscript
extends Control

var progress : TextureProgress

func _ready():
	progress = $TextureProgress

func _process(delta):
	progress.value += delta
	if (progress.value >= 1):
		progress.value = 0
```

**Comment by akien-mga** [comment]
Apparently `delta` is `0` in this script when using C# in the HTML export.

```csharp
using Godot;
using System;

public class Progress : Control
{
	private TextureProgress progress;

	public override void _Ready()
	{
		progress = GetNodeOrNull<TextureProgress>("TextureProgress");
	}

	public override void _Process(float delta)
	{
		GD.Print(delta); // 0 on web, 0.01667 on desktop
		progress.Value += delta;
		if(progress.Value >= 1){
			progress.Value = 0;
		}        
	}
}
```

CC @neikeq @Faless

**Comment by thes3m** [comment]
@akien-mga Didn't even notice that `delta` is `0`. Seems like there is a bigger issue, probably something is wrong with `float` types, seems like they are "rounded" to 0 or something. Setting `progress.Value += 0.016f` still does not work.

**Comment by akien-mga** [comment]
I've tried to narrow the regression down, so far I have:

- 3.2.4 beta 4: OK
- 3.2.4 RC 1: broken

I'll test other intermediate betas and update.

**Comment by akien-mga** [comment]
I haven't tried to bisect yet between 3.2.4 beta 4 and beta 5, will do in a bit.

But some potential candidates:

- Mono: Make Godot provide its own WASM m2n trampolines: https://github.com/godotengine/godot/pull/44373
- Mono: Add extra WASM framework assemblies on game export: https://github.com/godotengine/godot/pull/44374
- Fix Variant conversion to float instead of double: https://github.com/godotengine/godot/pull/44306

**Comment by akien-mga** [comment]
I can confirm that it's a regression from #44373, CC @neikeq.

I did two tests builds with the current build containers (same as for 3.3 RC 9) using Mono 6.12.0.122, and where the patch for WASM m2n trampolines should be properly applied:
https://downloads.tuxfamily.org/godotengine/testing/issue47898/

That's builds of https://github.com/godotengine/godot/commit/80c9f2d975a3d0c6dd10e36e0cf85bde5357c3da (merge of #44373) and https://github.com/godotengine/godot/commit/9710a38637a3456cac6dd902c14fa0346d260e0e (parent commit).
For both builds I had to also cherry-pick https://github.com/godotengine/godot/commit/bf942447a3bcb543832881093f4eac11fa97ebc8 to fix exporting HTML5 projects with the BCL.

80c9f2d is `bad` and 9710a38 is `good`.
